src/dynamic-attributes-span-processor.ts

```diff
@@ -0,0 +1,31 @@
+import { Attributes, Span } from '@opentelemetry/api';
+import { SpanProcessor } from '@opentelemetry/sdk-trace-base';
+
+interface PageRouteSpanProcessorOptions {
+  getAttributes: () => Attributes;
+}
+
+/**
+ * A {@link SpanProcessor} that adds dynamic attributes to every span
+ */
+export class DynamicAttributesSpanProcessor implements SpanProcessor {
```

Comment:
@pkanal, I started thinking that maybe a `PageRouteSpanProcessor` makes less sense than something more generic, so here is a `DynamicAttributesSpanProcessor` which can be exposed and the `BrowserAttributesSpanProcessor` can be implemented in terms of. This would avoid needing to introduce a separate processor for each little attribute that comes up over time.

Reply:
I like this approach! I'm wondering whether this should be something we package with the SDK or whether this should be documented for users since everyone's use case will be slightly different? 🤔 

Thoughts @wolfgangcodes / @MustafaHaddara ?

Reply:
I'm arriving at the same conclusion. I think adding dynamic attributes globally like this will be a common use case, but at the same time this is so exceedingly generic as to be almost worthless. I'm hoping that sometime soon the OTel client instrumentation SIG comes up with a proposal to allow dynamic resource attributes which might fit these use cases better. At that point the `page.route` thing is something you'd just set up in your render cycle with a `trace.addResourceAttributes({ 'page.route': match.route })` and would also take care of things like changing the `enduser.id` or `tenant.id` when a user logs in during a session.

Reply:
Hmmmm yeah, I _think_ that is what span processors are meant to do in general. So are we coming to the conclusion that the best thing to do here is make sure that this distro is able to support multiple span processors and that this should be expressed as a custom span processor in a consuming app?

Reply:
Yes, agreed, I'll close this PR